core/src/main/java/io/ddf/content/SchemaHandler.java

```diff
@@ -76,12 +76,12 @@ public int getNumColumns() {
   }
 
   @Override
-  public int getColumnIndex(String columnName) {
+  public int getColumnIndex(String columnName) throws DDFException {
     return mSchema != null ? mSchema.getColumnIndex(columnName) : -1;
```

Comment:
Actually, -1 is never returned because the schema's getColumnIndex throws Exception for non-existing columns already :D